packages/app/cypress/e2e/integration/runs.spec.ts

```diff
@@ -1,80 +1,249 @@
-describe('App: Runs Page', () => {
+import type { Interception } from '@packages/net-stubbing/lib/external-types'
+
+describe('App: Runs', () => {
   beforeEach(() => {
     cy.scaffoldProject('component-tests')
     cy.openProject('component-tests')
     cy.startAppServer()
   })
 
-  it('resolves the runs page', () => {
-    cy.loginUser()
-    cy.visitApp()
-    cy.wait(1000)
-    cy.get('[href="#/runs"]').click()
-    cy.get('[data-cy="runs"]')
+  context('Runs Page', () => {
+    it('resolves the runs page', () => {
+      cy.loginUser()
+      cy.visitApp()
+      cy.wait(1000)
+      cy.get('[href="#/runs"]').click()
+      cy.get('[data-cy="runs"]')
+      cy.get('[data-cy="app-header-bar"]').findByText('Runs').should('be.visible')
+    })
+
+    it('shows the loader', () => {
+      cy.loginUser()
+      cy.remoteGraphQLIntercept(async (obj) => {
+        await new Promise((resolve) => setTimeout(resolve, 200))
+
+        return obj.result
+      })
+
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.get('[data-cy="runs-loader"]')
+      cy.get('[data-cy="runs"]')
+    })
   })
 
-  it('shows the loader', () => {
-    cy.loginUser()
-    cy.remoteGraphQLIntercept(async (obj) => {
-      await new Promise((resolve) => setTimeout(resolve, 200))
+  context('Runs - Login', () => {
+    it('when logged out, shows call to action', () => {
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.contains('Log In').should('exist')
+    })
 
-      return obj.result
+    it('clicking the login button will open the login modal', () => {
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.contains('Log In').click()
+      cy.get('[id="headlessui-dialog-title-13"]').contains('Log in to Cypress')
+      cy.get('button').contains('Log In')
     })
 
-    cy.visitApp()
-    cy.get('[href="#/runs"]').click()
-    cy.get('[data-cy="runs-loader"]')
-    cy.get('[data-cy="runs"]')
+    it('if logged in and connected', { viewportWidth: 1200 }, () => {
+      cy.loginUser()
+      cy.visitApp()
+      cy.intercept('mutation-ExternalLink_OpenExternal', { 'data': { 'openExternal': true } }).as('OpenExternal')
+
+      cy.get('[href="#/runs"]').click()
+      cy.contains('a', 'OVERLIMIT').click()
+      cy.wait('@OpenExternal')
```

Comment:
You could use the [validateExternalLink](https://github.com/cypress-io/cypress/blob/10.0-release/packages/frontend-shared/cypress/e2e/support/e2eSupport.ts#L355) command to assert the presence of the link and its action on click with one-liner. Though you'll need to know your full link name... feel free to tweak the command if it makes sense to do so.

Reply:
I'm having some difficulty using this since part of the link name is a timestamp (HH:MM:SS) which changes every run. It is difficult to predict what the exact timestamp will be within the test which leads the link to not be found

Reply:
Ah yeah that's tricky. We could update the [stubbed run values](https://github.com/cypress-io/cypress/blob/10.0-release/packages/frontend-shared/cypress/support/mock-graphql/stubgql-CloudTypes.ts#L146) to return consistent dates; we'd have to do that if we ever wanted to do virtual regression here anyway. But yeah that's not necessary just to use the helper.